RecoTracker/TkMSParametrization/interface/MSLayer.h

```diff
@@ -16,9 +16,15 @@ class MSLayer {
   typedef PixelRecoRange<float> Range;
 
   struct DataX0 {
-    DataX0(const MSLayersKeeper* al = nullptr) : hasX0(false), allLayers(al) {}
+    DataX0(const MSLayersKeeper* al = nullptr) : hasX0(false), hasFSlope(false), allLayers(al) {}
     DataX0(float ax0, float asX0D, float aCotTheta)
-        : hasX0(true), hasFSlope(false), x0(ax0), sumX0D(asX0D), cotTheta(aCotTheta), allLayers(nullptr) {}
+        : hasX0(true),
+          hasFSlope(false),
+          x0(ax0),
+          sumX0D(asX0D),
+          cotTheta(aCotTheta),
+          slopeSumX0D(0.),
```

Comment:
Out of curiosity @Dr15Jones : I understand why one needs to initialize ```hasFSlope``` in the other constructor.
I wonder instead why do you need to add the initialization of ```slopeSumX0D``` here, or alternatively why you did not initialize all ```float```'s in the other constructor.

Reply:
>I wonder instead why do you need to add the initialization of slopeSumX0D here,

In the aarch64 builds the compiler is generating a warning about `slopeSumX0D` possibly being used uninitialized when that constructor is used.

>why you did not initialize all float's in the other constructor.

The compiler is not complaining about that constructor. (So laziness?)

Reply:
See https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc7_aarch64_gcc820/CMSSW_11_0_X_2019-10-08-2300/RecoTracker/TkMSParametrization